core/client/fs/src/test/java/alluxio/client/file/BaseFileSystemTest.java

```diff
@@ -447,4 +450,39 @@ public void unmountException() throws Exception {
       assertSame(EXCEPTION, e);
     }
   }
+
+  /**
+   * Ensures Warnings are logged or exceptions are thrown when an {@link AlluxioURI} with a
+   * scheme://host is passed.
+   */
+  @Test
+  public void UriCheckTest() throws Exception {
+    Configuration.set(PropertyKey.MASTER_HOSTNAME, "localhost");
+    Configuration.set(PropertyKey.MASTER_RPC_PORT, "19998");
+    AlluxioURI uri = new AlluxioURI("alluxio://localhost:1234/root");
+    boolean exceptionRaised = true;
+    try {
+      BaseFileSystem.UriCheck(uri);
+      exceptionRaised = false;
+      fail("Should have failed on bad host and port");
+    } catch (IllegalArgumentException e) {
+      assertTrue(exceptionRaised);
```

Comment:
good to assert something about `e.getMessage()` so that we know we're catching the right thing. You can check a substring with something like

`assertThat(e.getMessage(), containsString("some substring"));`

If that check fails, it will give a nice error message showing what `e.getMessage()` actually was.